ddtrace/profiling/collector/stack.pyx

```diff
@@ -311,29 +311,68 @@ cdef stack_collect(ignore_profiler, thread_time, max_nframes, interval, wall_tim
         if task_id in thread_id_ignore_list:
             continue
 
-        # Inject the task frame and replace the thread frame: this is especially handy for gevent as it allows us to
-        # replace the "hub" stack trace by the latest active greenlet, which is more interesting to show and account
-        # resources for.
-        if task_frame is not None:
-           frame = task_frame
-
-        frames, nframes = _traceback.pyframe_to_frames(frame, max_nframes)
-
-        event = StackSampleEvent(
-            thread_id=thread_id,
-            thread_native_id=thread_native_id,
-            thread_name=thread_name,
-            task_id=task_id,
-            task_name=task_name,
-            nframes=nframes, frames=frames,
-            wall_time_ns=wall_time,
-            cpu_time_ns=cpu_time,
-            sampling_period=int(interval * 1e9),
-        )
+        # Inject wall time for all running tasks
+        if thread_id == nogevent.main_thread_id:
+            for task_id, task_name, task_frame in _task.list_tasks():
+                if task_id == compat.main_thread.ident:
+                    # task_frame is None, so use the thread frames list here
+                    frames, nframes = _traceback.pyframe_to_frames(frame, max_nframes)
+
+                    event = StackSampleEvent(
+                        thread_id=thread_id,
+                        thread_native_id=thread_native_id,
+                        thread_name=thread_name,
+                        task_id=task_id,
+                        task_name=task_name,
+                        nframes=nframes, frames=frames,
+                        wall_time_ns=wall_time,
+                        cpu_time_ns=cpu_time,
+                        sampling_period=int(interval * 1e9),
+                    )
+
+                    # FIXME: we only trace spans per thread, so we assign the span to the main thread for now
```

Comment:
So this means the profiling information is not entirely correct, but at least the reported wall time is?

Reply:
Yes, basically the span is going to be assigned to the MainThread (which is a task). That might not be accurate since we do span <-> thread linking but we don't do span <-> task-in-thread linking.

(note: this is not a new problem, this problem already exists. I've just added this comment as it's clearer now that we do this explicitely)